core/commands/refs.go

```diff
@@ -171,21 +144,47 @@ Displays the hashes of all local objects.
 			return
 		}
 
-		piper, pipew := io.Pipe()
+		out := make(chan interface{})
+		res.SetOutput((<-chan interface{})(out))
 
 		go func() {
-			defer pipew.Close()
+			defer close(out)
 
 			for k := range allKeys {
-				s := k.B58String() + "\n"
-				if _, err := pipew.Write([]byte(s)); err != nil {
-					log.Error("pipe write error: ", err)
-					return
-				}
+				out <- &RefWrapper{Ref: k.B58String()}
 			}
 		}()
+	},
+	Marshalers: refsMarshallerMap,
+	Type:       RefWrapper{},
+}
+
+var refsMarshallerMap = cmds.MarshalerMap{
+	cmds.Text: func(res cmds.Response) (io.Reader, error) {
+		outChan, ok := res.Output().(<-chan interface{})
+		if !ok {
+			return nil, u.ErrCast()
+		}
+
+		marshal := func(v interface{}) (io.Reader, error) {
+			obj, ok := v.(*RefWrapper)
+			if !ok {
+				fmt.Println("%#v", v)
```

Comment:
I think it isn't needed, it might be a left over from someone debugging something.
It got here when I was rearranging the code.